scripts/Engineering/gui/engineering_diffraction/tabs/fitting/data_handling/data_model.py

```diff
@@ -384,6 +329,27 @@ def get_last_added(self):
     def get_sample_log_from_ws(self, ws_name, log_name):
         return self._loaded_workspaces[ws_name].getSampleDetails().getLogData(log_name).value
 
+    def _convert_TOF_to_d(self, tof, ws_name):
+        difa, difc, tzero = self._get_diff_constants(ws_name)
+        if abs(difa) < 1E-10:
+            return (tof - tzero) / difc
+        else:
+            return (-difc + sqrt(difc ** 2 - 4 * difa * (tzero - tof))) / (2 * difa)
+
+    def _convert_TOFerror_to_derror(self, tof_error, d, ws_name):
+        difa, difc, _ = self._get_diff_constants(ws_name)
+        return tof_error / (2 * difa * d + difc)
+
+    def _get_diff_constants(self, ws_name):
+        """Subject to change when ws will be able to carry diff constant on ws object post calibration"""
+        ws = ADS.retrieve(ws_name)
+        si = ws.spectrumInfo()
+        m_over_h = 252.816
+        difc = 2 * m_over_h * (si.l1() + si.l2(0)) * sin(si.twoTheta(0) / 2)
```

Comment:
Could we add some comments for people with a comp sci background (like me :wink: ) as to:
- What is a diff constant / what is it used for
- What is m_over_h
- What is the difc calculation were doing - is there a paper or reference? (unless it's something very standard in the Physics world, like we don't have to explain calculating error, in which case leave the comment out)

Reply:
Another good point, fortunately PR #30163 has also given us a way of directly accessing the diffractometer constants on a workspace so I will update to use this (no need for `m_over_h`)

Reply:
Ah it looks like not everything I need has been exposed to python  - as this is due to be updated soon I have just added comments as suggested.